src/WebJobs.Script.WebHost/Security/KeyManagement/BlobStorageSecretsRepository.cs

```diff
@@ -151,17 +160,23 @@ public override async Task<string[]> GetSecretSnapshots(ScriptSecretsType type,
             // Prefix is secret blob path without extension
             string prefix = Path.GetFileNameWithoutExtension(GetSecretsBlobPath(type, functionName)) + $".{ScriptConstants.Snapshot}";
 
-            BlobResultSegment segmentResult;
+            var blobList = new List<BlobItem>();
             try
             {
-                segmentResult = await Container.ListBlobsSegmentedAsync(string.Format("{0}/{1}", _secretsBlobPath, prefix.ToLowerInvariant()), null);
+                var segmentResult = Container.GetBlobsAsync(prefix: string.Format("{0}/{1}", _secretsBlobPath, prefix.ToLowerInvariant()));
+                var asyncEnumerator = segmentResult.GetAsyncEnumerator();
+
+                while (await asyncEnumerator.MoveNextAsync())
+                {
+                    blobList.Add(asyncEnumerator.Current);
+                }
             }
             catch (Exception e)
             {
                 LogErrorMessage("list");
                 throw e;
             }
-            return segmentResult.Results.Select(x => x.Uri.ToString()).ToArray();
+            return blobList.Select(x => x.Name).ToArray();
```

Comment:
Can we add names to the list in the first place?

Reply:
The functionality of the secrets manager relies on the actual blob name. The path and extension are stripped out when using this return value of blob names. In the ReadAsync call, the path is built again. Changing from uri -> blobName shouldn't cause problems here.

Reply:
Sorry, I need to work on the clarity of my comments. What I was saying is that we can change the list to be a list of string and do ` blobList.Add(asyncEnumerator.Current.Name);`

Reply:
Ah, good call. Will do.